include/zephyr/device.h

```diff
@@ -1079,8 +1079,38 @@ __syscall int device_init(const struct device *dev);
 
 DT_FOREACH_STATUS_OKAY_NODE(Z_MAYBE_DEVICE_DECLARE_INTERNAL)
 
+/** @brief Expands to the full type. */
+#define Z_DEVICE_API_TYPE(_class) _CONCAT(_class, _driver_api)
+
 /** @endcond */
 
+/**
+ * @brief Wrapper macro for declaring device API structs inside iterable sections.
+ *
+ * @param _class The device API class.
+ * @param _name The API instance name.
+ */
+#define DEVICE_API(_class, _name) const STRUCT_SECTION_ITERABLE(Z_DEVICE_API_TYPE(_class), _name)
+
+/**
+ * @brief Macro that evaluates to a boolean that can be used to check if an
+ *        API pointer is located in the iterable section.
+ *
+ * @param _class The device API class.
+ * @param _api The API instance pointer.
+ *
+ * @return true if the API pointer is located in the type's iterable section.
+ */
+#define DEVICE_API_IS(_class, _api)                                                                \
```

Comment:
I can't help thinking we should at least have a wrapper just taking a `const struct device *` and an a class.

This would hide the `->api` internals for callers and be more in line with the existing device.h APIs.

Thoughts?

Reply:
I've updated `DEVICE_API_IS` and added `DEVICE_API_GET`.